api/src/main/java/graphql/nadel/normalized/ValueToVariableValueCompiler.java

```diff
@@ -0,0 +1,107 @@
+
+    static VariableValueWithDefinition normalizedInputValueToVariable(NormalizedInputValue normalizedInputValue) {
+        Object variableValue = null;
+        Object normalizedInputValueValue = normalizedInputValue.getValue();
+        if (normalizedInputValueValue instanceof ObjectValue) {
+            variableValue = toVariableValue((ObjectValue) normalizedInputValueValue);
+        }
+        if (normalizedInputValueValue instanceof ArrayValue) {
+            variableValue = toVariableValues(((ArrayValue) normalizedInputValueValue).getValues());
+        }
+        if (normalizedInputValueValue instanceof List) {
+            variableValue = toVariableValues((List) normalizedInputValueValue);
+        }
+        String varName = getVarName();
+        return new VariableValueWithDefinition(
+                variableValue,
+                VariableDefinition.newVariableDefinition()
+                        .name(varName)
+                        .type(TypeName.newTypeName(normalizedInputValue.getTypeName()).build())
+                        .build(),
+                VariableReference.newVariableReference().name(varName).build());
+    }
+
+    @NotNull
+    private static List<Object> toVariableValues(List<Value> values) {
+        ArrayList<Object> objects = new ArrayList<>();
+        for (Value value : values) {
+            objects.add(toVariableValue(value));
+        }
+        return objects;
+    }
+
+    private static Map<String, Object> toVariableValue(ObjectValue objectValue) {
+        HashMap<String, Object> map = new HashMap<>();
+        List<ObjectField> objectFields = objectValue.getObjectFields();
+        for (ObjectField objectField : objectFields) {
+            String objectFieldName = objectField.getName();
+            Value<?> objectFieldValue = objectField.getValue();
+            if (objectFieldValue instanceof ArrayValue) {
+                List<Object> objects = toVariableValues(((ArrayValue) objectFieldValue).getValues());
+                map.put(objectFieldName, objects);
+                continue;
+            }
+            map.put(objectFieldName, toVariableValue(objectFieldValue));
+        }
+        return map;
+    }
+
+    private static Object toVariableValue(Value value) {
+        if (value instanceof ObjectValue) {
+            return toVariableValue((ObjectValue) value);
+        }
+        if (value instanceof StringValue) {
+            return ((StringValue) value).getValue();
+        }
+        if (value instanceof FloatValue) {
+            return ((FloatValue) value).getValue();
+        }
+        if (value instanceof IntValue) {
+            return ((IntValue) value).getValue();
+        }
+        if (value instanceof BooleanValue) {
+            return ((BooleanValue) value).isValue();
+        }
+        return null; //todo
+    }
+
+    private static String getVarName() {
+        return "var_" + UUID.randomUUID().toString().replace("-", "_");
```

Comment:
this looks like overkill as we only need to be unique for this query. UUID.randomUUID() has also a non zero cpu cost.
Can we just use a counter?